snuba/optimize_tracker.py

```diff
@@ -0,0 +1,81 @@
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Optional, Set
+
+from snuba import state
+from snuba.state import rds
+
+
+class OptimizedPartitionTracker(ABC):
```

Comment:
I don't see why any of this code needs to exist here. The InMemoryTracker is only used in a test and we have redis in our test environment. I would say let's not write code we don't use

Reply:
> I don't see why any of this code needs to exist here. The InMemoryTracker is only used in a test and we have redis in our test environment. I would say let's not write code we don't use

I don't like the idea of using infrastructure components unnecessarily for testing purposes. That's why I had introduced the InMemoryTracker. But I agree that it could be just in the test file. Removed it from the code for now.

Reply:
I don't think the abstract class needs to exist here either. We do way too much Java-like interface design when it's not really necessary in python

Reply:
Removed the abstract class